src/check.rs

```diff
@@ -107,3 +107,28 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         Err(e) => Err(CheckError::Http(HttpError::Fetch(url.clone(), e))),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::{check_file_url, CheckContext};
+    use std::env;
+    use url::Url;
+
+    macro_rules! test_paths {
+        ($($name:ident $path:literal,)*) => {
+            $(#[test]
+            fn $name() {
+                let cwd = env::current_dir().unwrap();
+                let url_path = cwd.join($path);
+                let url = Url::from_file_path(url_path).unwrap();
+
+                check_file_url(&url, &CheckContext{ check_http: false}).unwrap();
+            })*
+        }
+    }
```

Comment:
Can you make this a function instead that takes `paths: &[&str]`? The tests aren't so slow the difference will be noticable and I think functions are easier to read.

Reply:
Alternatively, you could take only one argument and have a separate `#[test]` for each path.